draft-yoshino-wish-00.txt

```diff
@@ -321,6 +257,70 @@ Internet-DrafWiSH: A General Purpose Message Framing over B October 2016
    needed.  What contents are exchanged and in what encoding they are
    exchanged over WiSH are to be defined by the application layer.
 
+6.  Using WiSH over HTTP
+
+   Standard HTTP (REST) semantics should be followed, especially the
+   choice of the HTTP method.  Some HTTP semantics may not be
+   applicable, e.g.  Cache-Control, when the body is streamed.  However,
+   such limitation is not specific to WiSH.
+
+7.  WebSocket Compatibility
+
+   In this section, we discuss how to bridge WiSH and WebSocket (API and
+   protocol).
+
+7.1.  Subprotocol Negotiation
+
+   When layered over HTTP, WiSH optionally negotiates the subprotocol,
+   i.e. the protocol layered over WiSH, by using the Accept header and
+   Content-Type header of the underlying HTTP.  RFC 6455 negotiated the
+
+
+
+Yoshino & Zhu              Expires May 2, 2017                  [Page 5]
+
+Internet-DrafWiSH: A General Purpose Message Framing over B October 2016
+
+
+   subprotocol by exchanging the Sec-WebSocket-Protocol header.  WiSH
+   lists the offered subprotocols in the Accept header as follows:
+
+       Accept: application/web-stream; protocol=foo; q=1,
+           application/web-stream; protocol=bar; q=0.5
+
+   Following the "application/web-stream" media type, a parameter named
+   "protocol" can be specified with the subprotocol name as its value.
+   Multiple subprotocols are offered by listing multiple application/
+   web-stream media type with the protocol parameter with different
+   values.  The server chooses one subprotocol from the offered ones and
+   notifies the chosen subprotocol by including the Content-Type header
+   with the WiSH media type with the subprotocol as its value as
+   follows:
+
+       Content-Type: application/web-stream; protocol=foo
+
+   WiSH does not forbid use of asymmetric subprotocols as different
+   values can be used for the Content-Type header in the HTTP request
+   and HTTP response (see Section 6).  This includes WiSH and non-WiSH
+   media type combinations.  However, symmetric subprotocols MUST be
+   used where compatibility with WebSocket matters.  A client MAY use
+   the WiSH media type without the protocol parameter in the Content-
+   Type header of the HTTP request to leave the subprotocol not
+   determined, and start communication using the subprotocol chosen by
+   the server once the HTTP response is received.
+
+7.2.  Valid UTF-8 Requirement
+
+   In RFC 6455, endpoints were required to _Fail the WebSocket
+   Connection_ when they find that the byte stream is not a valid UTF-8
+   stream.  The contents of text messages of WiSH also MUST be a valid
+   UTF-8 stream.  However, WiSH endpoints are not required to do
+   anything when they see an invalid UTF-8 stream.  This provides more
```

Comment:
This reads odd. 

... However, WiSH endpoints are not required to valid UTF-8 frames. This provides more flexibility to server frameworks, which may choose to check UTF-8 validness from a JSON parser.